From a3106d8157cf6fffd1dfd755c366cb6b982c95a7 Mon Sep 17 00:00:00 2001
From: Seth Shelnutt <seth@tiledb.io>
Date: Fri, 14 Jul 2023 10:12:21 -0400
Subject: [PATCH] Fix order of azure static library linking

This solves the missing symbol error by linking the azure-storage-common
library before the azure-storage-blobs library.

Error solved:
```
libtiledb.so.2.16: undefined symbol: _ZN5Azure7Storage9_internal23UrlEncodeQueryParameterERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
```
---
 tiledb/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tiledb/CMakeLists.txt b/tiledb/CMakeLists.txt
index 8f391fa0c..4ae46b552 100644
--- a/tiledb/CMakeLists.txt
+++ b/tiledb/CMakeLists.txt
@@ -545,8 +545,8 @@ if (TILEDB_AZURE)
     endif()
     target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
             INTERFACE
-            Azure::azure-storage-blobs
             Azure::azure-storage-common
+            Azure::azure-storage-blobs
             Azure::azure-core
             LibXml2::LibXml2
             LibLZMA::LibLZMA
@@ -562,8 +562,8 @@ if (TILEDB_AZURE)
     endif()
     target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
             INTERFACE
-            Azure::azure-storage-common
             Azure::azure-storage-blobs
+            Azure::azure-storage-common
             Azure::azure-core
             LibXml2::LibXml2)
     if(WIN32)
-- 
2.40.1

